### DISCUSSION TITLE
Potential breakage in x86_64-elf-binutils: 'as' fails to assemble source that an upstream binary works with

### DISCUSSION THREAD

**Discussion by jmillikin** [Homebrew/discussions#354] [Everyday usage] [answered]
This appears to be a bad build in the Homebrew bottle, but since my system's configuration is a bit weird I don't know for sure that it's broken for anyone else.

Given the following `asm-rept.S` source file:
```asm
.rept (1000/10)
.endr
```

`as` from Homebrew fails to assemble it:
```
$ x86_64-elf-as -o asm-rept.o asm-rept.S
asm-rept.S: Assembler messages:
asm-rept.S:1: Error: found '
', expected: ')'
$ x86_64-elf-as --version
GNU assembler (GNU Binutils) 2.35.1
Copyright (C) 2020 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or later.
This program has absolutely no warranty.
This assembler was configured for a target of `x86_64-elf'.
$
```

The same binutils version from upstream, configured with `--target=x86_64-linux-gnu`, works as expected:
```
$ ~/.opt/binutils-2.35.1/bin/x86_64-linux-gnu-as -o asm-rept.o asm-rept.S
$ ~/.opt/binutils-2.35.1/bin/x86_64-linux-gnu-as --version
GNU assembler (GNU Binutils) 2.35.1
Copyright (C) 2020 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or later.
This program has absolutely no warranty.
This assembler was configured for a target of `x86_64-linux-gnu'.
$
```

**Comment by SMillerDev** [comment]
What do `brew config` and `brew doctor` say?

**Comment by jmillikin** [comment]
> What do `brew config` and `brew doctor` say?

`brew doctor` output is far too long to post here -- like I said, this particular machine is weird. I'm mostly hoping somebody with a "standard" Homebrew setup can try installing that package and running the command to verify it it passes/fails.

Output of `brew config`:
```
$ brew config
HOMEBREW_VERSION: 2.7.0-5-g9c1bdfe
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9c1bdfe22906d4022a1a4aefc9365505a907bf7a
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 69d43acedd0fa4d1808fa90c0de666e1ff8c0841
Core tap last commit: 2 days ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: /usr/bin/vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 12.0 build 1200
Git: 2.25.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 1.8.0_275
macOS: 10.15.7-x86_64
CLT: 11.5.0.0.1.1588476445
Xcode: 12.0.1
```

**Comment by gromgit** [comment, marked as answer]
@jmillikin, I just tested manual source builds (`configure --target=XYZ`)  on both Ubuntu 20.04 and Catalina against your code. The results are instructive:
`--target=` | Linux | macOS (gcc) | macOS (clang)
-----|-----|-----|-----
`x86_64-elf` | **FAIL** | **FAIL** | **FAIL**
`x86_64-linux-gnu` | PASS | PASS | PASS

Looks like you'll need to report this upstream, as it's definitely not a Homebrew issue.
